### DISCUSSION TITLE
Nuxt Layers Monorepo - VSCode auto-imports

### DISCUSSION THREAD

**Discussion by kleinpetr** [nuxt/nuxt#19963] [Questions]
Hi

I am playing with Nuxt Layers and facing some troubles, especially with auto imports inside layers 

## Steps to reproduce
1. clone this example https://github.com/kleinpetr/nuxt-layers
2. run `pnpm i` in the root (it runs `nuxt prepare` at the end)
3. open `nuxt.config.ts` (you should see the error as below)

![image](https://user-images.githubusercontent.com/17269309/228300353-4ead324a-c80f-481a-ba72-3ffd4a7533ad.png)

Also when I run `pnpm build` or even `nuxi prepare` the error message disappears for a while but then returns back as you can see on the gif below

![image](https://user-images.githubusercontent.com/17269309/227916981-120d08cd-f958-41f9-8e56-5016c0b2b601.gif)

I am using Volar TS Exntension with Takeover mode (I also tried enable builtin TS Language Extension and Disable Volar, but the result is still the same) 

Do you have any suggestion what should be the problem?  (in the `/app` folder everything works fine, but inside `/layers` it is not)

Thanks for help :pray: 

---
OS: Linux Fedora 37
VSCode: version 1.76.2

**Comment by FilipEIG** [comment]
Having the same issue on Windows.

**Comment by edumudu** [comment]
The message disappears when you run the command because the ts types are updated, then the vscode reloads the tsconfig making the message briefly disappear. Remember that nuxt works with generated types

I don't clone your repo, but looking at the file structure, its probably because you are running the `nuxi dev` command in the `/app` folder and the layer folder is not a child of this `/app` folder, the way the tsconfig is loaded in vscode is recursively upwards since the tsconfig generated is inside the `/app`  folder when you open the layer the types doest exist there

If you move the `/app`  folder to the root, or put a tsconfig extending the `/app/tsconfig.json`  in the root probably gonna work

PS: you can test what I am saying simply by running the dev command in the layer instead of the `/app`, probably the error will also disappear
